common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -41,19 +41,43 @@ public ConceptService(EntityManager entityManager) {
   public Slice<Concept> searchConcepts(String query,
       StandardConceptFilter standardConceptFilter, List<String> vocabularyIds,
       List<String> domainIds, int limit) {
+
+      //String[] keywords=query.split("\\s*(=>|,|\\s)\\s*");
+      String[] keywords=query.split("[,+\\s+]");
+      for(int i=0;i<keywords.length;i++){
+          String key=keywords[i];
+          if(key.length() < 3){
```

Comment:
Why < 3? Why are we enclosing things in quotes here?

Reply:
I saw that the match function does not care to match any words that are less than three characters. Enclosing it in quotes instead works good.

Reply:
Interesting. What does it do if you try a match on one or two character strings... return nothing?

There's probably a reason to be wary of searching on short strings like this, in that you wind up matching a very large number of concepts. Perhaps we shouldn't be doing this?

Does this logic allow for researchers passing in keywords in quotes if they want to do an exact match? (I think it's fine if it doesn't... but we probably need to escape quotes and pluses and things that MATCH looks for...)

Reply:
It ignores matching anything after these short characters.
For example: If the search keyword is "type 2 diabetes" and we make it "+type +2 +diabetes"
it returns nothing, but "+type "2" +diabetes" returns all the concepts having the word.

Reply:
We can change the function to not modify the keyword in case it has double quotes and let the match do exact match.

Reply:
OK. It will be interesting to see how this performs. I worry slightly about what happens when a researcher just types in "a" or whatever. :)

Reply:
On my front end , i don't call the api  until they type  3 chararters or whatever i have it set to. 

If it ends up bogging down the server we could just return error if search term  is too short.

Reply:
OK. Since the API is public we should assume anybody could call it. If we don't want to support searching on < 3 characters, we probably shouldn't bother with this logic.